pkgs/development/haskell-modules/configuration-nix.nix

```diff
@@ -636,4 +636,9 @@ self: super: builtins.intersectAttrs super {
   # need it during the build itself, too.
   cairo = addBuildTool super.cairo self.buildHaskellPackages.gtk2hs-buildtools;
   pango = disableHardening (addBuildTool super.pango self.buildHaskellPackages.gtk2hs-buildtools) ["fortify"];
+
+  # The DateZone xmobar plugin has the hardcoded path /usr/share/zoneinfo
+  xmobar = overrideCabal super.xmobar (drv: {
+    preConfigure = "sed -i -e s,/usr/share/zoneinfo,${pkgs.tzdata}/share/zoneinfo, src/Xmobar/Plugins/DateZone.hs";
```

Comment:
It seems like in nixpkgs, people try to use `substituteInPlace` instead of `sed` directly.

Could you change this to use `substituteInPlace`?  If you grep through nixpkgs, you should be able to find a bunch of examples of it being used.